### ISSUE TITLE
No 'Access-Control-Allow-Origin' header is present on the requested resource

### ISSUE THREAD

**Issue by djsg** [tus/tus-node-server#111]
embed tus-js-client in a web brower; the endpoint is http://127.0.0.1:9000, which tus-node-server runs at.
I got the following error. Strangely I run the same setting in another machine; it works.

2upload.js:328 POST http://127.0.0.1:9003/files/ 412 (Precondition Failed)
_createUpload @ upload.js:328
start @ upload.js:190
startUpload @ demo.js:137
tus-js:1 Failed to load http://127.0.0.1:9003/files/: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://127.0.0.1:6060' is therefore not allowed access. The response had HTTP status code 412.
2upload.js:328 POST http://127.0.0.1:9003/files/ 412 (Precondition Failed)
_createUpload @ upload.js:328
start @ upload.js:190
(anonymous) @ upload.js:153
setTimeout (async)
_this.options.onError @ upload.js:152
_emitError @ upload.js:212
_emitXhrError @ upload.js:207
xhr.onerror @ upload.js:316
error (async)

**Comment by rictorres** [comment]
that's a [cors](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) issue.

maybe take a look at https://github.com/tus/tus-node-server/blob/master/lib/Server.js.

also, are you sending an `origin` header from the client?

**Comment by djsg** [comment]
@rictorres 
It seems to be caused by browser cache. I used a few endpoints to test.
After clearing browser cache, the problem disappears.

Another question: does the current demo/server.js support multiple clients uploading at the same time?

**Comment by djsg** [comment]
I am studying the server code. I wonder what log message is kept.

**Comment by Acconut** [comment]
> I wonder what log message is kept.

I am sorry but I don't fully understand what you mean. Could you rephrase that sentence?

**Comment by djsg** [comment]
@Acconut 
Can you check this issue [https://github.com/tus/tus-node-server/issues/114](url)? My question is regarding to it.

**Comment by Acconut** [comment]
> So far, I understand that using --trace-warning can enable log message to print to console. Is it correct?

I don't think that's fully correct but I didn't try it out. The --trace-warning flag makes that stack traces are added to the warning message. However, this does not enable the tus-node-server specific logging output.

If you want to see the log, you have to set the DEBUG environment variable: `DEBUG=tus-node-server:*`. How to exactly do that depends on your OS and shell, so please look that up for yourself.

If you are using bash, you can enable output and redirect it to a file like so:
```
DEBUG=tus-node-server:* node server.js > log.txt
```

**Comment by kvz** [comment]
Just a quick tip to also log stderr, because I think `debug` switched to stderr logging in some version

```
DEBUG=tus-node-server:* node server.js > log.txt 2>&1
```
